rust/tableau_summary/src/twb/diff/dashboard.rs

```diff
@@ -0,0 +1,149 @@
+use serde::{Deserialize, Serialize};
+use crate::twb::diff::util::{ChangeState, DiffItem, DiffProducer};
+use crate::twb::summary::dashboard::{Dashboard, Zone};
+
+#[derive(Serialize, Deserialize, Default, PartialEq, Clone, Debug)]
+pub struct DashboardDiff {
+    pub status: ChangeState,
+    pub num_changes: usize,
+    pub name: DiffItem<String>,
+    pub title: DiffItem<String>,
+    pub thumbnail: DiffItem<Option<String>>,
+    pub sheets: Vec<DiffItem<String>>,
+    pub zones: ZoneDiff,
+}
+
+impl DashboardDiff {
+    fn update_num_changes(&mut self) {
```

Comment:
```suggestion
    fn calculate_num_changes(&mut self) {
```
It looks to me that this is always called when the changes are zero and then they are calculated. If that is the case, the `update` in the name is a bit confusing